src/metabase/models/pulse.clj

```diff
@@ -154,20 +154,6 @@
   [_pulse]
   [:name (serdes/hydrated-hash :collection) :created_at])
 
-(def ^:private ^:dynamic *automatically-archive-when-last-channel-is-deleted*
-  "Should we automatically archive a Pulse when its last `PulseChannel` is deleted? Normally we do, but this is disabled
-  in [[update-notification-channels!]] which creates/deletes/updates several channels sequentially."
-  true)
-
-(defn will-delete-channel
```

Comment:
LSP thought this was unused, it was actually used, but only in one place -- `metabase.models.pulse-channel`, which dynamically resolves it. I thought that was all a bit silly so I just moved it over there so we don't need to dynamically resolve things anymore.